sniff.js

```diff
@@ -65,6 +65,11 @@ define(["./features"], function (has) {
 		has.add("mac", dav.indexOf("Macintosh") >= 0);
 
 		has.add("msapp", parseFloat(dua.split("MSAppHost/")[1]) || undefined);
+
+		var wp = parseFloat(dua.split("Windows Phone ")[1]) || undefined;
+		if (wp) {
+			has.add("wp", wp);
+		}
```

Comment:
> It looks that has.add could avoid adding the entry if the value is undefined. @clmath?

Yes it could. OTOH it will add complexity to the has plugin and this is far from being a performance issue.
So I would rather stay with the current implementation but if other people think this should be done I could do it.